### PR TITLE
Attempt to expand secondary space with ununsed space from first space

### PR DESCRIPTION
**Pull request by skjolber**
If nothing fits into the secondary (remainder) space, expand it with free space from primary space, if possible. 

Also fixes a potential bug with the secondary space fitting making the primare space fitting unavailable due to the weight constraint in the brute for packager.

### CODE REVIEW

**Comment by skjolber** [comment]
@tyrcho if you have some free time, please review these changes.

File: core/src/main/java/com/github/skjolber/packing/Space.java [hunk 1]
```diff
@@ -130,5 +133,104 @@ public void copyFrom(int w, int d, int h, int x, int y, int z) {
...
+
+		int startY = space.getY();
+		int endY = startY + space.getDepth() - 1;
+	
+		return intersectsY(startY, endY);
+	}
+
+	public boolean intersectsY(int startY, int endY) {
+
+		if (startY <= y && y <= endY) {
```
Comment (tyrcho) [inline core/src/main/java/com/github/skjolber/packing/Space.java]: these `intersectsA` methods could be simplified to
```java
between(startY, y, endY) || between(startY, y + depth, endY);
```

with 
```java
static boolean between(int start, int value, int end) {
  return start <= value && value <= end;
}
```

Comment (tyrcho) [reply core/src/main/java/com/github/skjolber/packing/Space.java]: and even 
```java
static boolean intersects(int start, int end, int value, int distance) {
  return between(start, value, end) || between(start, value + distance, end);
}

public boolean intersectsX(Space space) {
		int startX = space.getX();
		int endX = startX + space.getWidth() - 1;

		return intersects(startX, endX, x, width);
}
```

**Comment by tyrcho** [comment]
> @tyrcho if you have some free time, please review these changes.

I had a quick look, apart from this small possible refactoring I did not notice anything ... 

Nice looking 3D graphs would be great to support the comments though !

**Comment by skjolber** [comment]
Thanks @tyrcho! I agree that a proper visualization is necessary for moving forward. Like this: 

https://github.com/chadiik/freight-packer

**Pull request merged**